Source/ablastr/fields/MultiFabRegister.H

```diff
@@ -0,0 +1,713 @@
+            bool remake = true,
+            bool redistribute_on_remake = true
+        )
+        {
+            return alloc_init<std::string const &>(
+                getExtractedName(name),
+                dir,
+                level,
+                ba,
+                dm,
+                ncomp,
+                ngrow,
+                initial_value,
+                remake,
+                redistribute_on_remake
+            );
+        }
+
+        /** Create an alias of a MultiFab (field)
+         *
+         * Registers a new name for an existing MultiFab (field) and optionally assigning
+         * a value.
+         *
+         * @param new_name new name
+         * @param alias_name owner name to alias
+         * @param level the MR level to represent
+         * @param initial_value the optional value to assign
+         * @return the newly aliased MultiFab
+         */
+        template<typename N, typename A>
+        amrex::MultiFab*
+        alias_init (
+            N new_name,
+            A alias_name,
+            int level,
+            std::optional<amrex::Real const> initial_value = std::nullopt
+        )
+        {
+            return alias_init<std::string const &, std::string const &>(
+                getExtractedName(new_name),
+                getExtractedName(alias_name),
+                level,
+                initial_value
+            );
+        }
+
+        /** Create an alias of a MultiFab (field)
+         *
+         * Registers a new name for an existing MultiFab (field) and optionally assigning
+         * a value.
+         *
+         * @param new_name new name
+         * @param alias_name owner name to alias
+         * @param dir the field component for vector fields ("direction" of the unit vector) both in the alias and aliased
+         * @param level the MR level to represent
+         * @param initial_value the optional value to assign
+         * @return the newly aliased MultiFab
+         */
+        template<typename N, typename A>
+        amrex::MultiFab*
+        alias_init (
+            N new_name,
+            A alias_name,
+            Direction dir,
+            int level,
+            std::optional<amrex::Real const> initial_value = std::nullopt
+        )
+        {
+            return alias_init<std::string const &, std::string const &>(
+                getExtractedName(new_name),
+                getExtractedName(alias_name),
+                dir,
+                level,
+                initial_value
+            );
+        }
+
+        /** Allocate a new MultiFab (field) with the same size and distribution as another.
+         *
+         * @todo this is reserved for future use and not yet implemented.
+         * @todo we will also add an overload with dir.
+         *
+         * @param new_name new name
+         * @param other_name the other multifab
+         * @param level the MR level to represent
+         */
+        void
+        alloc_like (
```

Comment:
It seems that this function is unused / untested in WarpX. I would simply remove it.

Reply:
Ok